### PR TITLE
Feature #1797 Database optimization v2

### PR DESCRIPTION
**Pull request by PaulBender**
All Database optimization from #1797

PostgreSQL and SQLite don't support unsigned ints.

### CODE REVIEW

File: install/resources/upgrade33.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	flush();
+
+	switch($db->type)
+	{
+		// PostgreSQL and SQLite do not support unsigned ints
+		case "sqlite":
+			$db->modify_column("usergroups", "type", "tinyint NOT NULL default '2'");
+			break;
+		default:
+			$db->modify_column("usergroups", "type", "tinyint(2) unsigned NOT NULL default '2'");
```
Comment (Destroy666x) [inline install/resources/upgrade33.php]: PostgreSQL doesn't support tinyint.

File: install/resources/upgrade33.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+	$output->print_header("Updating Database");
+	echo "<p>Performing necessary optimization queries...</p>";
+	flush();
+
+	switch($db->type)
+	{
+		// PostgreSQL and SQLite do not support unsigned ints
+		case "sqlite":
+			$db->modify_column("usergroups", "type", "tinyint NOT NULL default '2'");
```
Comment (Destroy666x) [inline install/resources/upgrade33.php]: SQLite supports the number of digits in parentheses.

**Comment by PaulBender** [comment]
Made all changes.
-Changed type to tinyint(1) unsigned
-Dropped all unsigned references in SQLite install file
-Set a default of 0 for uid, touid and date in the buddyrequests table

**Comment by PaulBender** [comment]
Changed loginattempts to an unsigned smallint (http://community.mybb.com/thread-166156.html)

**Comment by PaulBender** [comment]
Fixed merged conflicts

**Pull request closed without merging**